lib/tile-analyze.js

```diff
@@ -44,102 +45,169 @@ class TileAnalyzeStream extends Transform {
 
 /**
  * Returns stats about an MBTiles file.
+ * Check if the tilestats object has been pre-generated in the mbtiles file.
+ * If so, return that, otherwise generate it with analyzeSourceStream.
  *
  * @param {string} filePath
  * @param {Object} [options]
  * @param {Array<string>} [options.attributes]
+ * @param {boolean} [options.intoMd]
  * @return {Object} The stats.
  */
-module.exports = function(filePath, options) {
-  options = options || {};
-  const stats = createStats();
-  const layerMap = {};
+module.exports = async function tileAnalyze(filePath, options) {
+  const source = await getSource(filePath);
+
+  const info = await getMetadata(source);
+  if (info.tilestats && validator(info.tilestats)) {
+    if (options.intoMd) throw new Error('Tilestats already exists, use without --into-md');
+    return info.tilestats;
+  }
+
+  const tilestats = await analyzeSourceStream(source, options);
+
+  if (options.intoMd) {
+    const report = reportStats(tilestats, options);
+    const metadata = { tilestats: report };
+    if (info.vector_layers) metadata.vector_layers = info.vector_layers;
```

Comment:
There could be two scenarios, even though just in theory:
- there is missing `json` record in metadata completely
- there is `json` record in metadata but it does not have `vector_layers` in it. 

Could we print two different warnings for these two cases?

Reply:
I am also getting this error with MBTiles with no `json` record in metadata table:
```
./bin/mapbox-geostats ~/data/mbtiles/cartiqo3.mbtiles --into-md
Missing vector_layers in metadata json!
TypeError: Cannot read properties of undefined (reading 'size')
    at module.exports (/home/adam/dev/maptiler/mapbox-geostats/lib/report-stats.js:22:37)
    at buildGeoStats (/home/adam/dev/maptiler/mapbox-geostats/index.js:32:10)
```

Reply:
Ok, I see the tool expects `size` from `vector_layers` in https://github.com/maptiler/mapbox-geostats/blob/CD-2725_insert-stats/lib/report-stats.js#L22
Maybe missing `json` or `vector_layers` should end up with error not just warning.

Reply:
I'm using `getInfo` function from mapbox mbtiles library and I am unable to detect if the whole json row is missing or just vector_layers. https://github.com/mapbox/node-mbtiles/blob/master/lib/mbtiles.js#L260-L298
I'd have to query the DB to distinguish this, which I'd prefer not to do.

The second thing `TypeError: Cannot read properties of undefined (reading 'size')` is really weird, can you send me `cartiqo3.mbtiles` to investigate it?

Reply:
Ok, it's a kind of edge-case, do not make difference between missing `json` or `vector_layers`.

Reply:
You can test it on any mbtiles and just run `REPLACE INTO metadata ('json', '') WHERE name='json';`
Or you can use these mbtiles: https://drive.google.com/file/d/1WdJIVGx3L-ECY2Rha7-1ywChfyqqvDNL/view?usp=drive_link

Reply:
Hmm, still cannot reproduce the error with undefined `size` you mention above. 
Tried different scenarios and it behaves as expected (crashing on @mapbox/mbtiles library when parsing).

`json` row missing, writes into created json row with a warning:
```
Missing vector_layers in metadata json!
```
---
`json` is set to `NULL`:
```
/home/vladan/dev/mapbox-geostats/node_modules/@mapbox/mbtiles/lib/mbtiles.js:271
                Object.keys(jsondata).reduce(function(memo, key) {
                       ^
TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at /home/vladan/dev/mapbox-geostats/node_modules/@mapbox/mbtiles/lib/mbtiles.js:271:24
    at Array.forEach (<anonymous>)
    at Statement.<anonymous> (/home/vladan/dev/mapbox-geostats/node_modules/@mapbox/mbtiles/lib/mbtiles.js:262:24)
```
---
`json` is set to empty string:
```
SyntaxError: Unexpected end of JSON input
    at JSON.parse (<anonymous>)
    at /home/vladan/dev/mapbox-geostats/node_modules/@mapbox/mbtiles/lib/mbtiles.js:269:43
    at Array.forEach (<anonymous>)
    at Statement.<anonymous> (/home/vladan/dev/mapbox-geostats/node_modules/@mapbox/mbtiles/lib/mbtiles.js:262:24)
```
---
`json` is set to `''`:
```
SyntaxError: Unexpected token ' in JSON at position 0
    at JSON.parse (<anonymous>)
    at /home/vladan/dev/mapbox-geostats/node_modules/@mapbox/mbtiles/lib/mbtiles.js:269:43
    at Array.forEach (<anonymous>)
    at Statement.<anonymous> (/home/vladan/dev/mapbox-geostats/node_modules/@mapbox/mbtiles/lib/mbtiles.js:262:24)
```

Reply:
Oh, yes, you are right with the mbtiles I've provided I get the same error `SyntaxError: Unexpected end of JSON input`.
The original cartiqo3.mbtiles files is quite big (over 7GB). I will reduce its size and upload it to GDrive.

Reply:
Nice catch! The error was caused by layerCount check. If the layerCount was zero, it behaved wrong.